File: Data Structures & Algorithms/anagram-groups/submission-0.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        res = defaultdict(list) #Charconnt:List of words
        for s in strs:
            count = [0] * 26 #a to z

            for c in s:
                count[ord(c) - ord("a")] += 1 #correcly sets letter count to be a-z

            res[tuple(count)].append(s)
        return list(res.values())

File: Data Structures & Algorithms/daily-temperatures/submission-0.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        int[] output = new int[temperatures.length];
        for(int i = 0; i < temperatures.length; i++){
            int count = 1;
            int j = i + 1;
            while(j < temperatures.length){
                if (temperatures[j] > temperatures[i]){
                    break;
                }
                j++;
                count++;
            }
            count = (j==temperatures.length) ? 0 : count;
            output[i] = count;
        }
        return output;
        
    }
}
